src/com/toopher/openam/ToopherSecondFactor.java

```diff
@@ -73,154 +68,80 @@ private void debug_message(String message) {
 
     @Override
     public void init(Subject subject, Map sharedState, Map options) {
-        super.init(subject, sharedState, options);
+        if (isUseFirstPassEnabled()) {
+            this.sharedState = sharedState;
+        }
+        if (sharedState == null) {
+            debug_message("sharedState is null!");
+        }
+        if (options == null) {
+            debug_message("options is null");
+        }
+        this.options = options;
+
+        String authLevel = CollectionHelper.getMapAttr(options, KEY_AUTHLEVEL);
+        if (authLevel != null) {
+            try {
+                setAuthLevel(Integer.parseInt(authLevel));
+            } catch (Exception e) {
+                debug.error("ToopherSecondFactor.init() : " + "Unable to set auth level " + authLevel, e);
+            }
+        }
+
+        String toopherConsumerKey = CollectionHelper.getMapAttr(options, KEY_CONSUMER_KEY);
+        String toopherConsumerSecret = CollectionHelper.getMapAttr(options, KEY_CONSUMER_SECRET);
+        String toopherApiUrl = CollectionHelper.getMapAttr(options, KEY_API_URL);
+
+        api = new ToopherIframe(toopherConsumerKey, toopherConsumerSecret, toopherApiUrl);
+
+        userName = (String) sharedState.get(getUserKey());
```

Comment:
No, it shouldn't be `null`.  The checks above are remnants of initial development when I didn't really know what to expect from them (the documentation for this stuff is not exactly stellar)